addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -211,6 +288,27 @@ function factory(dependencies) {
         checkedMessages: many2many('mail.message', {
             inverse: 'checkedThreadCaches',
         }),
+        /**
+         * List of messages that have been fetched by this cache.
+         *
+         * This DOES NOT necessarily includes all messages linked to this thread
+         * cache (@see messages field for that): it just contains list
+         * of successive messages that have been explicitly fetched by this
+         * cache. For all non-main caches, this corresponds to all messages.
+         * For the main cache, however, messages receive from longpolling should
+         * be displayed on main cache but they have not been explicitly fetched
+         * by cache, so they ARE NOT in this list (at least, not until a fetch
+         * on this thread cache contains this message).
+         *
+         * The distinction between messages and fetched messages is important
+         * to manage "holes" in message list, while still allowing to display
+         * new messages on main cache of thread in real-time.
+         */
+        fetchedMessages: many2many('mail.message', {
+            // adjust with messages linked to thread
```

Comment:
Wrong comment? I think it's just adjusting with messaged "unlinked" from thread, messaging linked to thread are not added on this compute.

Reply:
Comment meant "linked" as a general term, but indeed it adjusts from unlinked messages to thread.